addons/point_of_sale/models/res_partner.py

```diff
@@ -11,6 +11,11 @@ class ResPartner(models.Model):
         help="The number of point of sales orders related to this customer",
         groups="point_of_sale.group_pos_user",
     )
+    barcode = fields.Char(oldname='ean13', help="Use a barcode to identify this contact from the Point of Sale.", copy=False)
+
+    _sql_constraints = [
+        ('unique_barcode', 'unique(barcode, company_id)', 'This barcode is already assigned to another contact. Please make sure you assign a unique barcode to this contact.'),
```

Comment:
I will prefer this
`unique_barcode` --> `unique_barcode_company` or `barcode_company_unique` (as we have added two field in constraints)

Warning message: `This barcode is already assigned to another contact. Please make sure you assign a unique barcode to this contact.` -->  `Contact Barcode must be unique per company!`

Reply:
Hi @sswapnesh a contact barcode can be a simple loyalty card, so changing it to company is not a good idea.

Reply:
But constraints is applied for `barcode` and `company_id` both.

Reply:
This is for multi-company odoo databases. The constraint should be like that.

Reply:
Yes I know that, That's why I suggested to improve warning message and better constraints name.

Reply:
The message is already approved in functional review, so we are not going to change it any more. I think talking about companies would also be very confusing for customers that do not make use of multi company and don't know the code behind the check.